client.go

```diff
@@ -661,6 +662,10 @@ func (s *Client) LoopOutQuote(ctx context.Context,
 	// If we use an Asset we'll rfq to get the asset amounts to use for
 	// the swap.
 	if request.AssetRFQRequest != nil {
+		if s.assetClient == nil {
```

Comment:
we could wrap this and the `if` before into something like `isValidAssetRFQRequest`